src/action/config/ActionConfiguration.cpp

```diff
@@ -226,26 +221,25 @@ void ActionConfiguration::createAction()
   // Determine data and mesh
   std::vector<int> sourceDataIDs;
   int              targetDataID = -1;
-  mesh::PtrMesh    mesh;
-  for (const mesh::PtrMesh &aMesh : _meshConfig->meshes()) {
-    if (aMesh->getName() == _configuredAction.mesh) {
-      mesh = aMesh;
-      for (const mesh::PtrData &data : mesh->data()) {
-        if (std::find(_configuredAction.sourceDataVector.begin(), _configuredAction.sourceDataVector.end(), data->getName()) != _configuredAction.sourceDataVector.end()) {
-          sourceDataIDs.push_back(data->getID());
-        }
-        if (data->getName() == _configuredAction.targetData) {
-          targetDataID = data->getID();
-        }
-      }
-    }
-  }
-  PRECICE_CHECK(mesh,
+  PRECICE_CHECK(_meshConfig->hasMeshName(_configuredAction.mesh),
                 "Data action uses mesh \"{}\" which is not configured. Please ensure that the correct mesh name is given in <action:python mesh=\"...\">", _configuredAction.mesh);
+  mesh::PtrMesh mesh = _meshConfig->getMesh(_configuredAction.mesh);
+
+  if (!_configuredAction.targetData.empty()) {
+    PRECICE_CHECK(mesh->hasDataName(_configuredAction.targetData),
+                  "Data action uses target data \"{}\" which is not configured. Please ensure that the target data name is used by the mesh", _configuredAction.targetData);
+    targetDataID = mesh->data(_configuredAction.targetData)->getID();
+    PRECICE_ASSERT(targetDataID != -1);
+  }
+
+  for (const std::string &dataName : _configuredAction.sourceDataVector) {
+    PRECICE_CHECK(mesh->hasDataName(dataName), "Data action uses source data \"{}\" which is not configured. Please ensure that the target data name is used by the mesh", dataName);
```

Comment:
Could you include the name of the mesh?

Reply:
Done in a2a93f459cdf7ed97bf7b308f4693f7947968e06